packages/amino/src/signdoc.ts

```diff
@@ -69,5 +69,21 @@ export function makeSignDoc(
 }
 
 export function serializeSignDoc(signDoc: StdSignDoc): Uint8Array {
-  return toUtf8(sortedJsonStringify(signDoc));
+  const fixedMsgs = signDoc.msgs.map((aminoMsg) => {
+    if (aminoMsg.type.startsWith("regen")) {
+      // regen ledger incorrectly implements amino signing.
+      // this short term workaround makes cosmjs compatible with regen-ledger v4.0
+      // until regen ledger upgrades with the fix in v5.0
+      // ref: https://github.com/regen-network/regen-ledger/pull/1480
+      return aminoMsg.value
+    } else {
+      return aminoMsg
+    }
+  })
+
+  
+  var fixedSignDoc: any = signDoc;
+  fixedSignDoc.msgs = fixedMsgs;
+
+  return toUtf8(sortedJsonStringify(fixedSignDoc));
 }
```

Comment:
Testing your cosmjs fix against regen-js MsgSend unit test…
Now I get an error very similar to the one I get when I test regen-web:
`Amino type identifier 'undefined' does not exist in the Amino message type register. If you need support for this message type, you can pass in additional entries to the AminoTypes constructor. If you think this message type should be included by default, please open an issue at https://github.com/cosmos/cosmjs/issues`

Reply:
This makes me think the `type` is needed. Or maybe we have to modify something else in cosmjs?

Reply:
I think likely we need to change this somewhere else in the codepath then.

Reply:
Fixed in f5a802188985c0b661d86c5c445c16b0ee8baba5